projects/rocprofiler-systems/docs/how-to/openmp-profiling.rst

```diff
@@ -0,0 +1,198 @@
+
+.. image:: ../data/openmp-profiling/perfetto-jacobi-trace-details.png
+    :alt: Zoomed in view of the jacobi track showing details for the first ``omp_target_emi`` track
+    :width: 800
+
+Visualizing data transfer and kernel execution on the GPU
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``laplacian`` subroutine in ``laplacian.f90`` contains the following block of code that offloads compute to the GPU:
+
+.. code-block:: fortran
+
+  !$omp target teams distribute parallel do collapse(2)
+  do j = 2,mesh%n_y-1
+      do i = 2,mesh%n_x-1
+      au(i,j) = (-u(i-1,j)+2._RK*u(i,j)-u(i+1,j))*invdx2 &
+              + (-u(i,j-1)+2._RK*u(i,j)-u(i,j+1))*invdy2
+      end do
+  end do
+
+.. note::
+    This specific kernel code is executed many times. To find it in your trace, look for the corresponding ``__omp_offloading`` trace in the ``GPU Kernel Dispatch`` track that contains ``laplacian``
+    in its name (for example: ``__omp_offloading_10302_8e8155f__QMlaplacian_modPlaplacian_l21.kd``)
+
+The image below shows a group of traces that correspond to the execution of the block above:
+
+.. image:: ../data/openmp-profiling/perfetto-jacobi-laplacian-traces-view.png
+    :alt: Traces corresponding to the Laplacian OpenMP pragma
+    :width: 1400
+
+The general sequence of events for this code block is as follows:
+  1. An ``omp_target_emi`` callback is generated and spans the entire duration of the OpenMP ``target teams`` construct.
+  2. Memory is allocated on the GPU for variables. This is represented by an ``omp_target_data_op_emi`` trace with ``optype = target_data_alloc``.
+  3. The variables are then transferred to the GPU. This is represented by an ``omp_target_data_op_emi`` trace with ``optype = target_data_transfer_to_device``.
+  4. A corresponding ``MEMORY_COPY_HOST_TO_DEVICE`` is generated in the ``GPU Memory Copy to Agent`` track for each occurrence of (3).
+  5. Once the necessary data transfers are complete, the code can be executed on the kernel. An ``omp_target_submit_emi`` trace is generated and points to the kernel being executed on the GPU.
+  6. A corresponding ``__omp_offloading`` kernel is generated on the ``GPU Kernel Dispatch`` track for each occurrence of (5), representing the GPU code being executed.
+  7. Once complete, the data is transferred back to the host. This is represented by an ``omp_target_data_op_emi`` trace with ``optype = target_data_transfer_from_device``.
+  8. A corresponding ``MEMORY_COPY_DEVICE_TO_HOST`` is generated in the ``GPU Memory Copy to Agent`` track for each occurrence of (7).
+  9. The previously allocated memory is deallocated from the GPU. This is represented by an ``omp_target_data_op_emi`` trace with ``optype = target_data_delete``.
+
+In general, if a trace directly relates to another trace, an arrow will be generated between the two. These arrows are called "flow events". A flow event is visible only when a trace is selected.
```

Comment:
Saying "a trace directly relates to another trace" is confusing. A trace is the whole file with all the tracks. Are you referring to the events in the various tracks found in the trace? If so, clarify this here. Similarly, say "a flow event is visible only when an event on a track is selected".